includes/admin/class-rop-admin.php

```diff
@@ -1003,6 +1005,76 @@ public function rop_dismiss_cron_disabled_notice() {
 
 	}
 
+	/**
+	 * Migrate the taxonomies from General Settings to Post Format for Pro users.
+	 * @since 8.5.4
+	 */
+	public function migrate_taxonomies_to_post_format() {
+
+		$global_settings = new Rop_Global_Settings();
+		// If any type of Pro is installed and active.
+		if ( $global_settings->license_type() > 0 ) {
+			// Get the current plugin options.
+			$option = get_option( 'rop_data' );
+			// Get the custom options.
+			// If this option exists, then the migration took place, and it will not happen again.
+			// Should return false the first time as it does not exist.
+			$update_took_place = get_option( 'rop_data_migrated_tax' );
+
+			$general_settings = array();
+			// Making sure the option we need, exists.
+			if ( empty( $update_took_place ) && isset( $option['general_settings'] ) ) {
```

Comment:
I think it would be cleaner to return earlier and have the rest of the operation taking place outside of the if statement

So you can check for the $update_took_place option and check if $option['general_settings'] is set and if not we can just return

```
if ( ! empty( $update_took_place ) && ! isset( $option['general_settings'] ) ) {
return;
}
```

Reply:
I can change if you want, but it's the same as I did it. PHP knows not to run anything from the if statement and just skips it.